src/rulelib.cpp

```diff
@@ -281,7 +284,7 @@ ascii_to_vector(char *line, size_t len, int *nsamples, int *nones, VECTOR *ret)
 	ones = 0;
 
 
-	for(p = line; len-- > 0; p++) {
+	for(p = line; len-- > 0 && *p != '\0'; p++) {
```

Comment:
What if len is > 0 but *p == '\0'? Then we'd have a bug and probably want to throw an error, no?

Reply:
Yeah, for some reason that bug was present in some data files after compiling without GMP so I added that condition. I spend a bunch of time but simply couldn't find the bug. Fortunately, like this, if the data file actually has a problem it will be handled by line 318. Specifically, if one line is too short or empty, the loop will terminate early, and 'i' will be less than '*nsamples', triggering an error and causing the program to fail.